test-data/unit/check-literal.test

```diff
@@ -2945,3 +2945,88 @@ class C(B[Literal["word"]]):
 reveal_type(C().collection)  # N: Revealed type is "builtins.list[Literal['word']]"
 reveal_type(C().word)  # N: Revealed type is "Literal['word']"
 [builtins fixtures/tuple.pyi]
+
+[case testLiteralNarrowingInExpr]
+from typing import Any
+from typing_extensions import Literal
+
+def narrow(x: int, y: str, z: Literal[1, 2, 3]):
+    if x in (1, 2, "a", "b"):
+        reveal_type(x)                       # N: Revealed type is "Union[Literal[1], Literal[2]]"
+    else:
+        reveal_type(x)                       # N: Revealed type is "builtins.int"
+    if y in ("a", "b", 1, 2):
+        reveal_type(y)                       # N: Revealed type is "Union[Literal['a'], Literal['b']]"
+    else:
+        reveal_type(y)                       # N: Revealed type is "builtins.str"
+
+class A:
+    def __eq__(self, other: object) -> bool:
+        return True
+
+def no_narrow(a: int, b: int, c: Any, d: A):
+    if a in (1, 2, b):
+        reveal_type(a)                       # N: Revealed type is "builtins.int"
+    if c in (1,):
+        reveal_type(c)                       # N: Revealed type is "Any"
+    if d in (1, 2):
+        reveal_type(d)                       # N: Revealed type is "__main__.A"
+[builtins fixtures/ops.pyi]
```

Comment:
I had no clue which fixtures to use, so just chose something that seemed to work.